### PR TITLE
Task/39 manage game settings

### LINKED ISSUE #39: Manage the game settings
- Update database with new fields
- Update GameService
- Manage settings via socket

### PR DESCRIPTION
**Pull request by ThunderDev1**
- Added settings to db model.
- Update API to return game settings.
- Add SignalR method to update the settings.
- Wire settings component with update function.
- Minor optimization: return game data in refresh signal instead of making an API call.
- Commented service worker to make the project work in Firefox.

Closes #39

### CODE REVIEW

File: client/src/types/Settings.ts [hunk 1]
```diff
@@ -2,6 +2,6 @@ import { Difficulty } from './Difficulty'
 
 export interface Settings {
   difficulty: Difficulty
-  roundsToWin: number
-  cardSpeed: number
+  roundsToWin: Number
```
Comment (bellangerq) [inline client/src/types/Settings.ts]: What's the difference? 🤔 

In the [Typescript doc](https://www.typescriptlang.org/docs/handbook/basic-types.html#number), they write it in lowercase.

File: client/src/pages/Lobby/Lobby.tsx [hunk 2]
```diff
@@ -50,20 +50,36 @@ const Lobby: FC<LobbyProps> = (props: LobbyProps) => {
...
+              roundsToWin={game.roundsToWin}
+              cardSpeed={game.cardSpeed}
+              updateSettings={(settings: SettingsType) =>
+                hubConnection.invoke('UpdateGameSettings', settings)
+              }
+            />
+          ) : (
+            <ul>
+              <li>
+                Niveau de difficulté: <b>{game.difficulty}</b>
```
Comment (bellangerq) [inline client/src/pages/Lobby/Lobby.tsx]: Ok boomer.

File: client/src/pages/Lobby/Lobby.tsx [hunk 3]
```diff
@@ -50,20 +50,36 @@ const Lobby: FC<LobbyProps> = (props: LobbyProps) => {
...
+              <li>
+                Nombre de manches pour gagner: <b>{game.roundsToWin}</b>
+              </li>
+              <li>
+                Temps entre chaque carte: <b>{game.cardSpeed}</b> secondes
+              </li>
+            </ul>
+          )}
+
+          <div className='PlaisGameHost(game, player.id)yers'>
```
Comment (bellangerq) [inline client/src/pages/Lobby/Lobby.tsx]: Uh?

File: client/src/pages/Lobby/Lobby.tsx [hunk 4]
```diff
@@ -31,14 +31,14 @@ const Lobby: FC<LobbyProps> = (props: LobbyProps) => {
       setGame(game)
 
       if (game.status === 'PENDING_START') {
```
Comment (bellangerq) [inline client/src/pages/Lobby/Lobby.tsx]: Don't we need an enum for the game status?

File: client/src/pages/Lobby/Lobby.tsx [hunk 5]
```diff
@@ -2,21 +2,21 @@ import React, { FC, useEffect, useState } from 'react'
...
-
+import { Settings as SettingsType } from '../../types/Settings'
 import connectToGameHub from '../../utils/signalrConnector'
 import { Game } from '../../types/Game'
-import { Settings as SettingsType } from '../../types/Settings'
-import { Difficulty } from '../../types/Difficulty'
 import Settings from '../../containers/Settings'
 import './Lobby.scss'
 import PlayerBox from '../../components/PlayerBox'
+import { HubConnection } from '@microsoft/signalr'
```
Comment (bellangerq) [inline client/src/pages/Lobby/Lobby.tsx]: Split local and external imports:

externals
*< line break >*
locals

File: client/src/pages/Lobby/Lobby.tsx [hunk 6]
```diff
@@ -50,20 +50,36 @@ const Lobby: FC<LobbyProps> = (props: LobbyProps) => {
...
+          {isGameHost(game, currentPlayerId) ? (
+            <Settings
+              difficulty={game.difficulty}
+              roundsToWin={game.roundsToWin}
+              cardSpeed={game.cardSpeed}
+              updateSettings={(settings: SettingsType) =>
+                hubConnection.invoke('UpdateGameSettings', settings)
+              }
+            />
+          ) : (
```
Comment (bellangerq) [inline client/src/pages/Lobby/Lobby.tsx]: Can't the last part of the operator be directly in the `Settings` container? We would give it a `isHost` prop and render the correct content.

Or maybe another container? `HostSettings` and `PlayerSettings`? 🤔 

What do you think?

File: client/src/pages/Lobby/Lobby.tsx (hunk 2 above)
Comment (bellangerq) [reply client/src/pages/Lobby/Lobby.tsx]: It's best to use a `<strong>` tag instead. `<b>` tag is mostly used for styling purpose and HTML is not about styling things. `<strong>` has a semantic meaning.

**Comment by bellangerq** [comment]
Precise which issue it is closing.

File: client/src/pages/Lobby/Lobby.tsx (hunk 3 above)
Comment (ThunderDev1) [reply client/src/pages/Lobby/Lobby.tsx]: Well that's odd 😄

File: client/src/pages/Lobby/Lobby.tsx (hunk 6 above)
Comment (ThunderDev1) [reply client/src/pages/Lobby/Lobby.tsx]: Yes, absolutly. I prefer your second suggestion, having 2 components.

File: client/src/pages/Lobby/Lobby.tsx (hunk 5 above)
Comment (ThunderDev1) [reply client/src/pages/Lobby/Lobby.tsx]: I blame auto imports 😄

File: client/src/types/Settings.ts (hunk 1 above)
Comment (ThunderDev1) [reply client/src/types/Settings.ts]: Hmm I think you're right, I can't remember why I changed it.

**Pull request merged**